src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -177,30 +138,52 @@ private async Task RunLoop(CancellationToken cancellationToken)
                 );
 
             await Task.WhenAll(tasks.Select(s => s.Value));
-            _logger.LogTrace("... finished timeline");
 
-            var records = new List<AugmentedTimelineRecord>();
-            var issues = new List<AugmentedTimelineIssue>();
-            var augmentedBuilds = new List<AugmentedBuild>();
+            // Identify additional timelines by inspecting each record for a "PreviousAttempt"
+            // object, then fetching the "timelineId" field.
+            List<(Build build, Task<Timeline> timelineTask)> retriedTimelineTasks = new List<(Build, Task<Timeline>)>();
 
-            _logger.LogTrace("Aggregating results...");
             foreach ((Build build, Task<Timeline> timelineTask) in tasks)
             {
-                string targetBranch = "";
+                Timeline timeline = await timelineTask;
 
-                try
-                {
-                    if (build.Reason == "pullRequest")
-                    {
-                        targetBranch = (string) JObject.Parse(build.Parameters)["system.pullRequest.targetBranch"];
-                    }
-                }
-                catch (JsonReaderException e)
+                if (timeline is null)
                 {
-                    _logger.LogError(e.ToString());
+                    _logger.LogDebug("No timeline found for buildid {buildid}", build.Id);
+                    continue;
                 }
 
-                augmentedBuilds.Add(new AugmentedBuild(build, targetBranch));
+                IEnumerable<string> additionalTimelineIds = timeline.Records
+                    .Where(record => record.PreviousAttempts != null)
+                    .SelectMany(record => record.PreviousAttempts)
+                    .Where(attempt => attempt != null)
```

Comment:
Isn't this the same Where as 2 lines earlier?  Seems like one of the two of them can be omitted (I'd probably remove the first one)

Reply:
Indeed it is.